### ISSUE TITLE
mnist8m example doesn't work.

### ISSUE THREAD

**Issue by mrgloom** [JohnLangford/vowpal_wabbit#541]
I build vowpal wabbit under cygwin x32 on win7 x64.
to compile I change 
`boost_program_options-mt`
to 
`boost_program_options`
everywhere in makefiles

And it seems that only  `make mnist.test.confusion` works

`make mnist8m.test.confusion` fails because it can't find file to download

$ make mnist8m.test.confusion
wget http://ml.nec-labs.com/download/data/mnist8m/train8m-labels-idx1-ubyte.gz
SYSTEM_WGETRC = c:/progra~1/wget/etc/wgetrc
syswgetrc = C:\Program Files (x86)\GnuWin32/etc/wgetrc
--2015-03-01 10:23:11--  http://ml.nec-labs.com/download/data/mnist8m/train8m-labels-idx1-ubyte.gz
▒▒▒▒▒▒▒▒▒▒▒ ml.nec-labs.com... 138.15.110.15
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒▒▒▒ ▒ ml.nec-labs.com|138.15.110.15|:80... ▒▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒▒▒▒▒.
▒▒▒▒▒▒ HTTP ▒▒▒▒▒▒, ▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒... 404 Not Found
2015-03-01 10:23:12 ▒▒▒▒▒▒ 404: Not Found.

and everything with `png` just 'freeze' like this

$ make mnist11png.test.confusion
./do-mnist11png-train
final_regressor = mnist11png.model
Num weight bits = 20
learning rate = 0.05
initial_t = 0
power_t = 0.5
decay_learning_rate = 1
using input passthrough for neural network training
creating cache_file = mnist11png.cache
Reading datafile =
num sources = 1
average    since         example     example  current  current  current
loss       last          counter      weight    label  predict features

nothing happens, no cpu activity.

**Comment by arielf** [comment]
See: http://leon.bottou.org/projects/infimnist

_Unfortunately the original MNIST8M files have been deleted from the NEC servers. However you can use InfiMNIST to regenerate these files or generate much larger files if you prefer. You can even use this code to generate deformed MNIST examples on the fly._

Hope Paul can do something to fix this more permanently in the vw tree, but the above may provide some work-around.

**Comment by JohnLangford** [comment]
Yeah, this requires some tweaking.

-John

**Comment by JohnLangford** [comment]
Ping for Paul.

-John

**Comment by pmineiro** [comment]
When I get a fresh clean copy of master, and run 'make -n mnist8m.test.confusion' in the demo/mnist directory, I don't see any attempts to download the mnist8m files from bottou.org.  Is OP using an old build of vw?

-------------------------------------

pmineiro@pmineirovb-14% make -n mnist8m.test.confusion
test -x "$(which wget)" || {                            \
          echo "ERROR: you need to install wget" 1>&2;          \
          exit 1;                                               \
        }
wget http://leon.bottou.org/_media/projects/infimnist.tar.gz
tar xvfz infimnist.tar.gz
make -C infimnist
(cd infimnist && ./infimnist lab 10000 8109999) | gzip > train8m-labels-idx1-ubyte.gz
(cd infimnist && ./infimnist pat 10000 8109999) | gzip > train8m-images-idx3-ubyte.gz
test -x "$(which zsh)" || {                             \
          echo "ERROR: you need to install zsh" 1>&2;           \
          exit 1;                                               \
        }
test -x "$(which perl)" || {                            \
          echo "ERROR: you need to install perl" 1>&2;          \
          exit 1;                                               \
        }
test -x "$(which paste)" || {                           \
          echo "ERROR: you need to install paste" 1>&2;         \
          exit 1;                                               \
        }
./do-mnist8m-train
wget http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz
wget http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz
./do-test mnist8m
perl -lane '++$n; ++$c if $F[0] != $F[1]; ++$m{"@{[int($F[0])]}:$F[1]"}; } { print "mnist8m test errors: $c out of $n"; print "confusion matrix (rows = truth, columns = prediction):"; foreach $true (1 .. 10) { print join "\t", map { $m{"$true:$_"} || 0 } (1 .. 10); }' mnist8m.test.predictions

**Comment by JohnLangford** [comment]
Must be?  Closing, but reopen if not.
